jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -655,3 +655,28 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
             if self.subset_definitions[index][i]['name'] == name:
                 self.subset_definitions[index][i]['value'] = new_value
                 break
+
+    def _import_spectral_regions(self, spec_region, mode=NewMode, viewer=None):
+        """
+        Method for importing a SpectralRegion object or list of SpectralRegion objects.
+
+        Parameters
+        ----------
+        spec_region : ~`specutils.SpectralRegion` object or list
+            The object that contains bounds for the spectral region or regions
+        mode : AndMode, AndNotMode, OrMode, ReplaceMode, XorMode, NewMode, or list
+            By default this is set to NewMode which creates a new subset per subregion.
+            OrMode will create a composite subset with each subregion corresponding
+            to a subregion of a single subset.
+        """
+        viewer_name = viewer or self.app._jdaviz_helper._default_spectrum_viewer_reference_name
+        range_viewer = self.app.get_viewer(viewer_name)
+        for index, sub_region in enumerate(spec_region):
+            if isinstance(mode, list):
```

Comment:
Looks like this needs tests.

Also, isn't `len(mode)` supposed to be `len(spec_region) - 1`? Unless you want `spec_region[0]` to be able to be applied to whatever came before that (which this function has no knowledge of)?

Reply:
Tests added, let me know if that is sufficient.